File: 신고 결과 받기.py
def solution(id_list, report, k):
    answer = [0]*len(id_list)
    user = {x:[] for x in id_list}
    reported = {x:0 for x in id_list}
    
    for i in set(report):
        name = i.split()[0]
        re_name = i.split()[1]
        user[name].append(re_name)
        reported[re_name] += 1
        
    idx = 0
    for value in user.values():
        for i in value:
            if reported[i] >= k:
                answer[idx] += 1
        idx += 1
    
    return answer

File: 신고 결과 받기.js
function solution(id_list, report, k) {
  let answer = [];
  const obj = {};
  const count = {};
  const ansObj = {};
  for (let i = 0; i < id_list.length; i++) {
    count[id_list[i]] = 0;
    ansObj[id_list[i]] = 0;
    obj[id_list[i]] = [];
  }
  report.forEach((el) => {
    const [user, warn] = el.split(" ");
    if (!obj[user].includes(warn)) {
      count[warn] += 1;
    }
    obj[user] = [...obj[user], warn];
  });
  for (let i = 0; i < id_list.length; i++) {
    if (count[id_list[i]] >= k) {
      for (let key in obj) {
        if (obj[key].includes(id_list[i])) {
          ansObj[key]++;
        }
      }
    }
  }
  for (let key in ansObj) {
    answer.push(ansObj[key]);
  }
  console.log(obj);
  console.log(count);
  console.log(ansObj);
  return answer;
}
